docs/en/quality_en.md

```diff
@@ -0,0 +1,147 @@
+
+#### Content
+
+- **Function test**
+  
+  Function test is to conduct a comprehensive test on the functions provided by the module from user's prospective. Before the test, it is necessary to design sufficient test cases, considering various system states and parameter input to decide whether the module can still work as usual, and return the expected results. In this process, a scientific test cases design method is required to consider various input scenarios, and complete the test process with least cases and execution steps.
+  
+- **Exception test**
+  
+  Different from function test and performance test, exception test is another test for discovering performance bottlenecks caused by exceptions such as system exception, dependencies exception and the exception of the application itself in order to improve the stability of the system. Common exceptions include disk errors, network errors, data errors or program restarts, etc.  We use software to simulate exceptions in the development stage, such as "libfiu" in C++ and "gofail'' in go, which can simulate various exceptions.
+
+- **Scale test**
+  
+  This is to test whether the module can work normally under a certain scale, and whether it will crash or shows exceptions by observing the utilization of system resources. For example, test whether there will be errors when opening a large number of files and whether the memory usage will be too large.
+
+- **Concurrent/pressure test**
+
+  Functional test is more of a single-threaded test. It is also necessary to test the module in a concurrent scenario to observe whether it can still work normally, and whether there will be errors in logic or data. When considering the concurrency degree, we consider a larger pressure, for example, 2 times, 10 times or more than our normal scenario.
+
+#### Test case design
+
+Here we provide an [example of the test case for datastore module](quality-integration-example.md).
+
+### System test
+
+As mentioned at the beginning, the system test is a black box test. The system test of Curve usually conducted by QA. Tests include are: normal test, performance test, stability test, exception test and chaos test.
+
+- Normal test is basically manual test for newly added functions;
+
+- Performance test is for testing the performance;
+
+- Stability test requires the system to run for a certain amount of time under normal pressure
+
+- Anomaly test injects software and hardware exception into the normal process
+
+- Chaos test is to randomly combine software and hardware exceptions in normal process  
+
+
+Unit tests and integration tests are usually coded by developers in code development and can be accumulated continuously. In the process of system testing, we also hope to automate test cases as much as possible to reduce manual regression costs, so that test cases can continue to accumulate.
```

Comment:
in code development -> in development stage

Reply:
> in code development -> in development stage

done